includes/sanitizers/class-amp-nav-menu-toggle-sanitizer.php

```diff
@@ -99,6 +99,11 @@ public function sanitize() {
 		$script_el->appendChild( $this->dom->createTextNode( wp_json_encode( $expanded ) ) );
 		$state_el->appendChild( $script_el );
 		$nav_el->parentNode->insertBefore( $state_el, $nav_el );
+		if ( 'body' === $nav_el->nodeName ) {
+			$nav_el->insertBefore( $state_el, $nav_el->firstChild );
```

Comment:
I like `insertBefore` rather than `appendChild` because it puts the `amp-state` closer to the element attributes that are being modified.

Just note that `amp-auto-ads` has docs that state:

> The `<amp-auto-ads>` tag should be placed as the first child of the `<body>`.

The key word is _should_. It actually does not cause a validation error to have it elsewhere than the first child:

![image](https://user-images.githubusercontent.com/134745/57812049-bfadad80-7720-11e9-8f63-472b3fc83483.png)

This has been important for WordPress since before `wp_body_open` we've had to rely on `wp_footer` which has worked.

For `style[amp-keyframes]` it is also [specified](https://amp.dev/documentation/guides-and-tutorials/learn/spec/amphtml#keyframes-stylesheet):

> May only be placed as the last child of the document's `<body>` element.

This is an actual _hard_ error:

![image](https://user-images.githubusercontent.com/134745/57811983-8d03b500-7720-11e9-9fda-605fd6cbf46d.png)

All this to say, `insertBefore` like you have here is very correct.